### ISSUE TITLE
Bug: (Camlecase rule: problems with detecting missing camelcase)

### ISSUE THREAD

**Issue by Kvrl** [eslint/eslint#15191]
### Environment

Node version: 14.15.1
npm version: 6.14.9
Local ESLint version: 7.32.0 
Global ESLint version: Not Found
Operating System:  win32 10.0.19042

### What parser are you using?

@babel/eslint-parser

### What did you do?

<details>
<summary>Configuration</summary>

```
{
    "env": {
        "browser": true,
        "es2021": true
    },
    "parserOptions": {
        "parser": "babel-eslint",
        "ecmaVersion": 12,
        "sourceType": "module"

    },
    "rules": {
        "camelcase": ["error",{"properties": "never"}],
    }
}

```
</details>

```js
        methods: {
            bugLinter(){
                let test_var = 'Hello';
                let test = test_var;
                console.info(test_var);
                console.error(test_var);
                console.log(this.transform(test_var));
            },
            transform(msg){
                return `${msg} World`;
            }
        }
```

### What did you expect to happen?

That all five camelcase violations are shown when we run es lint.

### What actually happened?

Es-lint only reported two out of five violations.
It looks like es-lint is ignoring the vars in the brackets of the console.logs or in general method calls.

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

The given code is part of a vue component, just in case.

**Comment by nzakas** [comment]
Please provide the actual ESLint output you are seeing.

**Comment by mdjermanovic** [comment]
I think the intent was to allow function calls like `test_var()`, but the implementation mistakenly allows arguments like `foo(test_var)` as well, so this is most likely a bug.

That said, I'm not sure if we should change this behavior now. It doesn't seem important to report these references, because the declaration is already reported so the rule enforces the use of camelcase style for this variable anyways.

**Comment by nzakas** [comment]
Yeah, it doesn’t make sense to report read references, as that’s just noise. In VS Code you can press F2 and rename all references, and I’m sure other IDEs have similar functionality.

Closing as this is the intended behavior.
